tensorflow/python/kernel_tests/sparse_ops_test.py

```diff
@@ -743,5 +743,46 @@ def testMismatchedShapes(self):
         tf.sparse_maximum(sp_zero, sp_one).eval()
 
 
+class SparseTransposeTest(tf.test.TestCase):
```

Comment:
We'd definitely want tests for tensors of rank 0, 1, 2, and 3, and various value datatypes. I'd suggest just generating random tensors and write a generic test that is called or each, instead of hard-coded tests. Move to/merge with https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/transpose_op_test.py.